src/plugins/discover/public/application/services/use_es_doc_search.ts

```diff
@@ -37,7 +37,7 @@ export function buildSearchBody(
         },
       },
       stored_fields: computedFields.storedFields,
-      script_fields: computedFields.scriptFields,
```

Comment:
I don't think this is the right place to cast this. Both of those variables are typed and there is a type missmatch, meaning we're having somewhere not the right types. In this case it seems to me, that `DataView.getComputedField` should make sure that what it returns under `scriptFields` actually matches the type that's expected to put into a request for `script_fields`, i.e. I suggest something like this:


```patch
diff --git a/src/plugins/data/common/data_views/data_views/data_view.ts b/src/plugins/data/common/data_views/data_views/data_view.ts
index 596887e83ec..e49d9709568 100644
--- a/src/plugins/data/common/data_views/data_views/data_view.ts
+++ b/src/plugins/data/common/data_views/data_views/data_view.ts
@@ -10,6 +10,7 @@
 
 import _, { each, reject } from 'lodash';
 import { castEsToKbnFieldTypeName } from '@kbn/field-types';
+import type { estypes } from '@elastic/elasticsearch';
 import { FieldAttrs, FieldAttrSet, DataViewAttributes } from '../..';
 import type { RuntimeField } from '../types';
 import { DuplicateField } from '../../../../kibana_utils/common';
@@ -158,7 +159,7 @@ export class DataView implements IIndexPattern {
   };
 
   getComputedFields() {
-    const scriptFields: Record<string, { script: { source: string; lang: string } }> = {};
+    const scriptFields: Record<string, estypes.ScriptField> = {};
     if (!this.fields) {
       return {
         storedFields: ['*'],
@@ -185,7 +186,7 @@ export class DataView implements IIndexPattern {
       scriptFields[field.name] = {
         script: {
           source: field.script as string,
-          lang: field.lang as string,
+          lang: field.lang,
         },
       };
     });
diff --git a/src/plugins/discover/public/application/services/use_es_doc_search.ts b/src/plugins/discover/public/application/services/use_es_doc_search.ts
index c5216c483fd..a2f0cd6f844 100644
--- a/src/plugins/discover/public/application/services/use_es_doc_search.ts
+++ b/src/plugins/discover/public/application/services/use_es_doc_search.ts
@@ -37,7 +37,7 @@ export function buildSearchBody(
         },
       },
       stored_fields: computedFields.storedFields,
-      script_fields: computedFields.scriptFields as Record<string, estypes.ScriptField>,
+      script_fields: computedFields.scriptFields,
       version: true,
     },
   };
```